packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -2464,6 +2464,80 @@ describe("typespec-client-generator-core: decorators", () => {
         UsageFlags.JsonMergePatch | UsageFlags.Input | UsageFlags.Json
       );
     });
+
+    it("@usage Input and Output on Namespace", async () => {
+      const { OrphanModel, InputModel, OutputModel, RoundtripModel } = (await runner.compile(`
+        @service({})
+        @test
+        @usage(Usage.input | Usage.output)
+        namespace MyService {
+          @test
+          model OrphanModel {
+            prop: string;
+          }
+
+          @test
+          model InputModel {
+            prop: string
+          }
+
+          @test
+          model OutputModel {
+            prop: string
+          }
+
+          @test
+          model RoundtripModel {
+            prop: string
+          }
+
+          @route("/one")
+          op one(@body body: InputModel): OutputModel;
+
+          @route("/two")
+          op two(@body body: RoundtripModel): RoundtripModel;
+        }
+      `)) as { OrphanModel: Model; InputModel: Model; OutputModel: Model; RoundtripModel: Model };
+      strictEqual(getUsage(runner.context, OrphanModel), UsageFlags.Input | UsageFlags.Output);
+      // this is set to input and ouput because of the namespace override
+      strictEqual(
```

Comment:
@tadelesh can you confirm that the usage for `InputModel` and `OutputModel` should be both `Usage.input` and `Usage.output` because of the `@usage` decorator?

Reply:
yes. i think the `@usage` on ns override their usage.